### PR TITLE
reactphysics3d: init at 0.9.0

### PR DESCRIPTION
**Pull request by rexxDigital**
## Description of changes

Add ReactPhysics3d (https://github.com/DanielChappuis/reactphysics3d)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by Henry-Hiles — CHANGES_REQUESTED** [review]

File: pkgs/development/libraries/reactphysics3d/default.nix
Comment (eclairevoyant) [inline pkgs/development/libraries/reactphysics3d/default.nix:11]:
```suggestion
    hash = "sha256-AUdsUXsygsGfS8H+AHEV1fSrrX7zGmfsaTONYUG3zqk=";
```

**Review by eclairevoyant — CHANGES_REQUESTED** [review]

File: pkgs/development/libraries/reactphysics3d/default.nix
Comment (rexxDigital) [reply pkgs/development/libraries/reactphysics3d/default.nix]: > Other than that, lgtm if it builds/works, I'll add an approval once this is fixed.

builds, tested using nixpkgs-review

Comment (eclairevoyant) [reply pkgs/development/libraries/reactphysics3d/default.nix]: since you switched to the `finalAttrs` idiom, rather than using a let-expression, I'd suggest just using `rev = "v${finalAttrs.version}";`

Comment (rexxDigital) [reply pkgs/development/libraries/reactphysics3d/default.nix]: > since you switched to the `finalAttrs` idiom, rather than using a let-expression, I'd suggest just using `rev = "v${finalAttrs.version}";`

Will do, asked in discord didnt get an answer so i just put it in here to see what people would say!

**Review by eclairevoyant — APPROVED** [review]

File: pkgs/development/libraries/reactphysics3d/default.nix
Comment (NobbZ) [inline pkgs/development/libraries/reactphysics3d/default.nix:10]: I am generally not a friend of this.

This does not help overriding the version of a package, just confuse users.

Either move the version into a `let`, hardcode it in both locations or make it part of the `src`-drvs `name` (which is sadly discouraged).

Comment (NobbZ) [reply pkgs/development/libraries/reactphysics3d/default.nix:10]: In the latter case, there would be *some* error reporting, on overrding the version rather than just substituting the old sources (as it would happen now).

At the end, whatever you do, an explicit override of the `src` will always be required, and therefore using `finalAttrs` in the definition of `src` doesn't make sense.

**Review by Henry-Hiles — APPROVED** [review]
LGTM

File: pkgs/development/libraries/reactphysics3d/default.nix
Comment (SuperSandro2000) [reply pkgs/development/libraries/reactphysics3d/default.nix:10]: we should just use rec here

Comment (Henry-Hiles) [reply pkgs/development/libraries/reactphysics3d/default.nix:10]: Unfortunate, I'm the one who told them to use finalAttrs. As far as I know, it'd the preferred pattern

Comment (rexxDigital) [reply pkgs/development/libraries/reactphysics3d/default.nix:10]: Will change back to rec after work

File: pkgs/top-level/all-packages.nix
Comment (wegank) [inline pkgs/top-level/all-packages.nix:12581]: Please move the package to `pkgs/by-name/re/reactphysics3d/package.nix` and remove this line, thanks.

Comment (rexxDigital) [reply pkgs/top-level/all-packages.nix:12581]: May i ask why? Seems like e nitpick since it is a library

Comment (eclairevoyant) [reply pkgs/top-level/all-packages.nix:12581]: new top-level packages that use `pkgs.callPackage` should be in `by-name`

Comment (rexxDigital) [reply pkgs/top-level/all-packages.nix:12581]: > new top-level packages that use `pkgs.callPackage` should be in `by-name`

I see, so i should move it and remove it from top-level packages?

**Review by wegank — APPROVED** [review]

**Review by wegank — APPROVED** [review]

**Review by Henry-Hiles — APPROVED** [review]
LGTM

**Comment by Henry-Hiles** [comment]
Wait does this have 2 or 3 approvals

**Comment by rexxDigital** [comment]
> Wait does this have 2 or 3 approvals

Depends if @eclairevoyant still approves

**Pull request merged**
